target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8068-ap3935.dts

```diff
@@ -344,6 +344,10 @@
 
 	phy-mode = "sgmii";
 	phy-handle = <&phy2>;
+
+	nvmem-cells = <&ethaddr>;
+	nvmem-cell-names = "mac-address";
+	mac-address-increment = <1>;
```

Comment:
use the new nvmem-layout way.

Reply:
@Ansuel Sorry, I've looked around for an example of how to do this, but I might be missing what you're asking to do here. The value here comes from uboot-env nvmem driver, and I can't find any prior art for using the nvmem-layout with that driver. If you can give me a breadcrumb, I'm happy to go figure it out.

As far as I can tell uboot-env can only parse MAC addresses from the environment variable `ethaddr`, hence deriving the second MAC from the first (which is as the devices ship from the factory in all cases I've been able to check):

https://github.com/torvalds/linux/blob/d2da77f431ac49b5763b88751a75f70daa46296c/drivers/nvmem/u-boot-env.c#L125-L129

Reply:
We want to use `#nvmem-cell-cells` rather than `mac-address-increment`. Example:
https://github.com/openwrt/openwrt/blob/889a50e2660ecaabc33db2a3fa0e1edcd636f8de/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts#L223
https://github.com/openwrt/openwrt/blob/889a50e2660ecaabc33db2a3fa0e1edcd636f8de/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts#L382
`u-boot,env` [does support post processing](https://elixir.bootlin.com/linux/latest/C/ident/u_boot_env_read_post_process_ethaddr). OpenWrt backport: https://github.com/openwrt/openwrt/blob/main/target/linux/generic/backport-5.15/811-v6.4-0018-nvmem-u-boot-env-post-process-ethaddr-env-variable.patch
For now, `u-boot,env` is it's own driver, waiting for nvmem-layouts changes upstream before the u-boot nvmem driver might change to a nvmem layout driver…:
- https://lore.kernel.org/lkml/20231011111529.86440-5-miquel.raynal@bootlin.com/
- https://lore.kernel.org/lkml/20230718160652.26756-1-zajec5@gmail.com/t/#u

Reply:
Thank you—it makes sense now. This little bit of code was what I was missing (I think): https://github.com/torvalds/linux/blob/d2da77f431ac49b5763b88751a75f70daa46296c/drivers/nvmem/u-boot-env.c#L86-L87

Pushing shortly.

Reply:
@Ansuel Sorry if this is a dumb question, but because this is coming from u-boot-env do I need to add the mac-base compatible? I only see that used when the mac is parsed out of a fixed-layout... I don't see it in the nbg dts:

https://github.com/openwrt/openwrt/blob/c22aa0be3e2df4d46010fa57a1004c17c1288781/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8074-nbg7815.dts#L222-L224

Also it looks like during one of the rebases of #13030 the code which set the MAC after boot got deleted before it was merged... it used to be here in this older commit from that branch/PR: https://github.com/openwrt/openwrt/commit/1c210ed1fb78cf188f321a0e006a85c37d0a6db6#diff-62a0adc62a308675dc7632a28e801a5f73af12b35c7dd395bb22d1ecc0a4f456R108-R110

Reply:
For what it's worth, an image built from this branch builds and assigns the MAC addresses correctly.

Reply:
@rmilecki is it ok to define like that?

Reply:
To be clear, this comment thread is on a stale commit, please look at the current head of the branch—am on my phone or I'd link